mlrun/runtimes/nuclio/function.py

```diff
@@ -1327,3 +1327,23 @@ def get_nuclio_deploy_status(
     else:
         text = "\n".join(outputs) if outputs else ""
         return state, address, name, last_log_timestamp, text, function_status
+
+
+def enrich_nuclio_function_from_headers(
+    func,
+    headers,
```

Comment:
add type hints please

Reply:
@alonmr I tried to add `Union[RemoteRuntime, ApplicationRuntime]`, but then I got an error about circular imports..

Reply:
Added